mne/io/neuralynx/neuralynx.py

```diff
@@ -136,13 +273,7 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
             [len(segment.analogsignals) for segment in neo_block[0].segments]
         ) == len(neo_block[0].segments)
 
-        # collect sizes of each segment
-        segment_sizes = np.array(
-            [
-                nlx_reader.get_signal_size(0, segment_id)
-                for segment_id in range(len(neo_block[0].segments))
-            ]
-        )
+        segment_sizes = self._raw_extras[0]["segment_sizes"]
```

Comment:
This will matter if you ever concatenate multiple RawNeuralynx instances
```suggestion
        segment_sizes = self._raw_extras[fi]["segment_sizes"]
```
Looks like we need a similar fix to `exclude_filename` above, maybe also other places